src/services/audius-backend/Cognito.ts

```diff
@@ -0,0 +1,47 @@
+import AudiusBackend, {
+  AuthHeaders,
+  IDENTITY_SERVICE
+} from 'services/AudiusBackend'
+import { waitForLibsInit } from 'services/audius-backend/eagerLoadUtils'
+
+// @ts-ignore
+const libs = () => window.audiusLibs
+
+type HttpMethod = 'POST' | 'GET' | 'PUT' | 'DELETE' | 'PATCH'
+
+export type CognitoSignatureResponse = { signature: string }
+export type CognitoFlowResponse = { shareable_url: string }
+
+async function _makeRequest<T>({
+  path,
+  method = 'GET'
+}: {
+  path: string
+  method?: HttpMethod
+}): Promise<T> {
+  await waitForLibsInit()
+  const account = libs().Account.getCurrentUser()
+  if (!account) {
+    throw new Error('Cognito Identity Request Failed: Missing current user')
+  }
```

Comment:
i have seen this kinda check in a few places, waiting for libs, checking user, getting signdata for headers. not something we need to tackle in this pr, but curious if you feel like it would make sense to add some middlewares to capture this logic?